### PR TITLE
Use Language Server consistent java home to launch debuggee

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

It fixes the issue https://github.com/Microsoft/vscode-java-debug/issues/366.

This PR will fix the JDK instead of JRE issue, but not fix the parameter issue for JDK library, where the JDK method still shows `arg0, arg1, ...`. This is a known issue for JDK debugging. See the doc http://www.thejavageek.com/2016/04/03/debug-jdk-source-code/ and https://dev.zhourenjian.com/blog/2007/08/04/debug-into-rtjar.html

### CODE REVIEW

**Comment by testforstephen** [comment]
The peered Java side PR: https://github.com/Microsoft/java-debug/pull/225

File: src/utility.ts [hunk 1]
```diff
@@ -109,3 +113,25 @@ export function formatErrorProperties(ex: any): IProperties {
 
     return properties;
 }
+
+export function checkJavaHome(): Promise<string> {
```
Comment (andxu) [inline src/utility.ts]: getJavaHome

File: src/utility.ts [hunk 2]
```diff
@@ -109,3 +113,25 @@ export function formatErrorProperties(ex: any): IProperties {
...
+    return new Promise((resolve, reject) => {
+        let javaHome: string = readJavaConfig() || process.env.JDK_HOME || process.env.JAVA_HOME;
+        if (javaHome) {
+            javaHome = expandHomeDir(javaHome);
+            if (pathExists.sync(javaHome) && pathExists.sync(path.resolve(javaHome, "bin", `javac${EXE_SUFFIX}`))) {
+                return resolve(javaHome);
+            }
+        }
+
+        findJavaHome((err, home) => {
```
Comment (andxu) [inline src/utility.ts]: will findJavaHome expand the home dir like above expandHomeDir?

**Comment by testforstephen** [comment]
@yaohaizh Thanks for the concern. I created an issue on vscode-java about exporting the resolve java home info. See the issue link https://github.com/redhat-developer/vscode-java/issues/639

File: src/utility.ts (hunk 2 above)
Comment (testforstephen) [reply src/utility.ts]: nice catch.

**Pull request merged**